pyproject.toml

```diff
@@ -68,3 +68,6 @@ sphinx_rtd_theme = "*"
 [build-system]
 requires = ["setuptools","poetry>=0.12"]
 build-backend = "poetry.masonry.api"
+
+[tool.pytest.ini_options]
+filterwarnings = ["ignore::UserWarning:cvxpy", "ignore::DeprecationWarning:cvxpy"]
```

Comment:
To Do: Understand why the addition of `ignore::DeprecationWarning:cvxpy` did not cover the following warnings raised for a virtualenv. 

https://github.com/vprusso/toqito/actions/runs/6793475410/job/18468371615?pr=212#step:5:682
https://github.com/vprusso/toqito/actions/runs/6793475410/job/18468371615?pr=212#step:5:686

```
../../../.cache/pypoetry/virtualenvs/toqito-QfgmswsN-py3.10/lib/python3.10/site-packages/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py:136
  /home/runner/.cache/pypoetry/virtualenvs/toqito-QfgmswsN-py3.10/lib/python3.10/site-packages/cvxpy/reductions/solvers/conic_solvers/sdpa_conif.py:136: DeprecationWarning: invalid escape sequence '\i'
    """

../../../.cache/pypoetry/virtualenvs/toqito-QfgmswsN-py3.10/lib/python3.10/site-packages/cvxpy/reductions/cone2cone/soc2psd.py:49
  /home/runner/.cache/pypoetry/virtualenvs/toqito-QfgmswsN-py3.10/lib/python3.10/site-packages/cvxpy/reductions/cone2cone/soc2psd.py:49: DeprecationWarning: invalid escape sequence '\l'
```

Reply:
Was not fixed by https://github.com/vprusso/toqito/pull/212/commits/100f2af33465837b712284345d9574439b73dd2e where all warnings from `cvxpy` are ignored.

Reply:
We might want to keep these warnings in and not ignore them if they are telling us something important. It might be worthwhile to keep them in, especially if ignoring them proves difficult.

Reply:
Fine by me. 

Resolving this is related to how pytest runs in a particular directory. From what I found, people do try to skip the `virtualenvs`. But the things I tried so far are not letting me skip it. 

https://stackoverflow.com/questions/34651379/how-to-make-py-test-cov-skip-virtualenv-directory

One thing I figured out is cached versions of packages are being installed instead of the latest versions. Most of the `virtualenvs` warnings were removed by upgrading to a newer version of pytest-cov. `2.10.1` --> `4.1.0`